File: backend-v2/app/schemas/technical_case.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field, field_serializer


def _serialize_datetime(dt: datetime | None) -> str | None:
    if dt is None:
        return None
    return dt.isoformat() if isinstance(dt, datetime) else str(dt)


class TechnicalCaseSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    title: str
    primary_review_item: str
    secondary_review_item: str
    case_type: str
    scene_tags: str  # JSON string
    keywords: str    # JSON string
    summary: str
    contract_name: str
    contract_amount: str
    client_name: str
    score: str
    status: str
    created_at: datetime

    @field_serializer('created_at')
    def serialize_created_at(self, dt: datetime) -> str:
        return _serialize_datetime(dt) or ""


class TechnicalCaseDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    title: str
    primary_review_item: str
    secondary_review_item: str
    case_type: str
    scene_tags: str
    keywords: str
    summary: str
    contract_name: str
    contract_amount: str
    client_name: str
    contract_overview: str
    key_highlights: str
    content: str
    score: str
    status: str
    source: str
    video_url: str
    created_at: datetime
    updated_at: datetime

    @field_serializer('created_at', 'updated_at')
    def serialize_dt(self, dt: datetime) -> str:
        return _serialize_datetime(dt) or ""


class TechnicalCaseCreateRequest(BaseModel):
    title: str
    primary_review_item: str = ""
    secondary_review_item: str = ""
    case_type: str = "项目案例"
    scene_tags: list[str] = []
    keywords: list[str] = []
    summary: str = ""
    contract_name: str = ""
    contract_amount: str = ""
    client_name: str = ""
    contract_overview: str = ""
    key_highlights: str = ""
    content: str = ""
    source: str = ""


class TechnicalCaseUpdateRequest(BaseModel):
    title: str | None = None
    primary_review_item: str | None = None
    secondary_review_item: str | None = None
    case_type: str | None = None
    scene_tags: list[str] | None = None
    keywords: list[str] | None = None
    summary: str | None = None
    contract_name: str | None = None
    contract_amount: str | None = None
    client_name: str | None = None
    contract_overview: str | None = None
    key_highlights: str | None = None
    content: str | None = None
    status: str | None = None
    source: str | None = None
    video_url: str | None = None

File: frontend-v2/src/services/technicalCase.ts
import api from './api'

// ============ Technical Case (技术案例) ============

export interface TechnicalCaseSummary {
  id: string
  project_id: string
  title: string
  primary_review_item: string
  secondary_review_item: string
  case_type: string
  scene_tags: string  // JSON string
  keywords: string    // JSON string
  summary: string
  contract_name: string
  contract_amount: string
  client_name: string
  score: string
  status: string
  created_at: string
}

export interface TechnicalCaseDetail extends TechnicalCaseSummary {
  contract_overview: string
  key_highlights: string
  content: string
  source: string
  video_url: string
  updated_at: string
}

export interface CreateTechnicalCasePayload {
  title: string
  primary_review_item?: string
  secondary_review_item?: string
  case_type?: string
  scene_tags?: string[]
  keywords?: string[]
  summary?: string
  contract_name?: string
  contract_amount?: string
  client_name?: string
  contract_overview?: string
  key_highlights?: string
  content?: string
  source?: string
}

export async function listTechnicalCases(projectId: string): Promise<TechnicalCaseSummary[]> {
  const res = await api.get<TechnicalCaseSummary[]>(
    `/projects/${projectId}/technical-cases`
  )
  return res.data
}

export async function getTechnicalCaseDetail(
  projectId: string,
  caseId: string
): Promise<TechnicalCaseDetail> {
  const res = await api.get<TechnicalCaseDetail>(
    `/projects/${projectId}/technical-cases/${caseId}`
  )
  return res.data
}

export async function createTechnicalCase(
  projectId: string,
  payload: CreateTechnicalCasePayload
): Promise<TechnicalCaseDetail> {
  const res = await api.post<TechnicalCaseDetail>(
    `/projects/${projectId}/technical-cases`,
    payload
  )
  return res.data
}

export async function updateTechnicalCase(
  projectId: string,
  caseId: string,
  payload: Partial<CreateTechnicalCasePayload & { status?: string; video_url?: string }>
): Promise<TechnicalCaseDetail> {
  const res = await api.patch<TechnicalCaseDetail>(
    `/projects/${projectId}/technical-cases/${caseId}`,
    payload
  )
  return res.data
}

export async function deleteTechnicalCase(
  projectId: string,
  caseId: string
): Promise<void> {
  await api.delete(`/projects/${projectId}/technical-cases/${caseId}`)
}

export async function searchTechnicalCases(
  projectId: string,
  params: { primary_item?: string; secondary_item?: string; keyword?: string }
): Promise<TechnicalCaseSummary[]> {
  const res = await api.get<TechnicalCaseSummary[]>(
    `/projects/${projectId}/technical-cases-search/search`,
    { params }
  )
  return res.data
}